daemon/state.go

```diff
@@ -96,8 +96,11 @@ func (d *Daemon) validateEndpoint(ep *endpoint.Endpoint) (valid bool, err error)
 		return false, fmt.Errorf("no workload could be associated with endpoint")
 	}
 
-	if err := d.allocateIPsLocked(ep); err != nil {
-		return false, fmt.Errorf("Failed to re-allocate IP of endpoint: %s", err)
+	// If the IP is managed by an external IPAM, it does not need to be re-allocated
+	if !ep.DatapathConfiguration.ExternalIPAM {
```

Comment:
The comment above this check and this check seem to be opposites?

I think the logic is correct, the comment just confused me.